### PR TITLE
Switch from bootstrap-select to tom-select

### PR DESCRIPTION
**Pull request by nwalters512**
`bootstrap-select` doesn't fully support Bootstrap 5, which is a problem given we want to upgrade to Bootstrap 5 (#10095).

This PR switches us to use [`tom-select`](https://github.com/orchidjs/tom-select) instead, which better supports Bootstrap 5.

### CODE REVIEW

**Review by jonatanschroeder — APPROVED** [review]
A couple of accessibility nits for cases where a mouse/tap is not used:

* when using keyboard, tabbing into the element opens the dropdown. This is slightly different than a regular dropdown that doesn't open automatically, but it's not a big deal.
* once the element has focus, if there is no selected item yet, up/down don't do anything, I need to type the answer to get something, and can only get the item to be selected if I type enough for a single option to match my search.
* if there is a selected item, up/down go to the other options as expected, but hitting tab in one of the options does not select the option. I need to hit enter to actually select the item before changing focus to get the option to be selected. This deviates from the behaviour of a regular select.

I don't think these are reasons to block this PR, so I'm approving it, but they should be addressed eventually.

**Review by mylesw27 — APPROVED** [review]
Tested with questions from the example course and seemed to work well.

**Comment by nwalters512** [comment]
> when using keyboard, tabbing into the element opens the dropdown. This is slightly different than a regular dropdown that doesn't open automatically, but it's not a big deal.

Good catch, I changed this behavior in 4076140.

> once the element has focus, if there is no selected item yet, up/down don't do anything, I need to type the answer to get something, and can only get the item to be selected if I type enough for a single option to match my search.

This is a byproduct of poor handling of the initiall-selected-but-disabled option that we use as a placeholder of sorts. I did some monkeypatching in 4076140 to fix this.

> if there is a selected item, up/down go to the other options as expected, but hitting tab in one of the options does not select the option. I need to hit enter to actually select the item before changing focus to get the option to be selected. This deviates from the behaviour of a regular select.

AFAICT this is OS-dependent. On macOS, hitting `<tab>` when a regular `<select>` is open selects the first option but does not close the dropdown or accept the selected option. The ARIA `combobox` role doesn't specify that <kbd>Tab</kbd> should do anything at all: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role

@jonatanschroeder @mylesw27 could you retest the above changes and make sure everything looks and feels good to you?

**Comment by mylesw27** [comment]
> could you retest the above changes and make sure everything looks and feels good to you?

I had not tried the keyboard before, but it now seems to work great and function as expected when using the keyboard.

**Review by jonatanschroeder — APPROVED** [review]

File: apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.mustache [hunk 1]
```diff
@@ -59,27 +59,80 @@
 {{/radio}}
 {{^radio}}
 <!-- Weirdly enough putting this in a div causes it to display without the inline -->
-<span class="py-1 d-inline align-items-center">
+<span
+    class="py-1 d-inline align-items-center pl-multiple-choice-dropdown"
+    {{#size}}style="--pl-multiple-choice-dropdown-width: {{size}}px"{{/size}}
+>
     <script>
```
Comment (jonatanschroeder) [inline apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.mustache:66]: This script tag is now at ~60 lines with nothing other than the uuid as dynamic data. Maybe in a future PR, but i think it's worth moving this to a `pl-multiple-choice.js` file instead.

Comment (nwalters512) [reply apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.mustache:66]: I'll do it now 👍

Comment (nwalters512) [reply apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.mustache:66]: Done in 86983ba.

File: apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    },
+  });
+
+  // In case the dropdown items contain math, render it when the
+  // dropdown is opened or closed.
+  select.on('dropdown_open', () => {
+    // The first time the dropdown is opened, this even is fired before the
+    // options are actually present in the DOM. We'll wait for the next tick
+    // to ensure that the options are present.
+    setTimeout(() => MathJax.typesetPromise([container]), 0);
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.js:39]: I made this change when moving this all to a script file; I noticed that math wasn't rendering the first time I opened the dropdown when clicking.

File: apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+/* eslint-env browser */
+/* global TomSelect, MathJax */
+
+window.PLMultipleChoice = function (uuid) {
```
Comment (jonatanschroeder) [inline apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.js:4]: Really nit, but should this be `PLMultipleChoiceSelect`? On the off chance that we have a new variation of multiple choice in the future?

**Review by jonatanschroeder — APPROVED** [review]

File: apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.js (hunk 3 above)
Comment (nwalters512) [reply apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.js:4]: I had the same thought, but I will treat this as a nit and ignore it. It is entirely trivial to rename this in the future if we need to.

**Pull request merged**